src/main/java/com/activestate/cloudfoundryjenkins/CloudFoundryPushPublisher.java

```diff
@@ -196,156 +217,172 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
-            int running = 0;
-            int totalInstances = 0;
-            for (int tries = 0; tries < TIMEOUT; tries++) {
-                running = 0;
-                InstancesInfo instancesInfo = client.getApplicationInstances(app);
-                if (instancesInfo != null) {
-                    List<InstanceInfo> listInstances = instancesInfo.getInstances();
-                    totalInstances = listInstances.size();
-                    for (InstanceInfo instance : listInstances) {
-                        if (instance.getState() == InstanceState.RUNNING) {
-                            running++;
-                        }
-                    }
-                    if (running == totalInstances && totalInstances > 0) {
-                        break;
-                    }
-                }
-                Thread.sleep(1000);
-            }
-
-            String instanceGrammar = "instances";
-            if (running == 1)
-                instanceGrammar = "instance";
-            listener.getLogger().println(running + " " + instanceGrammar + " running out of " + totalInstances);
-
-            if (running > 0) {
-                if (running != totalInstances) {
-                    listener.getLogger().println("WARNING: Some instances of the application are not running.");
-                }
-                if (deploymentInfo.isNoRoute()) {
-                    listener.getLogger().println("Application is now running. (No route)");
-                } else {
-                    listener.getLogger().println("Application is now running at " + appURI);
-                }
-                listener.getLogger().println("Cloud Foundry push successful.");
-                return true;
-            } else {
-                listener.getLogger().println("ERROR: The application failed to start after " + TIMEOUT + " seconds.");
-                listener.getLogger().println("Cloud Foundry push failed.");
-                return false;
-            }
-        } catch (CloudFoundryException e) {
-            listener.getLogger().println("ERROR: Unknown CloudFoundryException: " + e.getMessage());
-            listener.getLogger().println("ERROR: Cloud Foundry error code: " + e.getCloudFoundryErrorCode());
-            if (e.getDescription() != null) {
-                listener.getLogger().println("ERROR: " + e.getDescription());
-            }
-            e.printStackTrace(listener.getLogger());
-            return false;
-        } catch (FileNotFoundException e) {
-            listener.getLogger().println("ERROR: Could not find file: " + e.getMessage());
-            return false;
-        } catch (ZipException e) {
-            listener.getLogger().println("ERROR: ZipException: " + e.getMessage());
-            return false;
-        }
-    }
+ BuildListener listener, URL targetUrl) throws IOException,
+			InterruptedException {
+		try {
+			String OrigAppURI = "https://" + deploymentInfo.getHostName() + "." + deploymentInfo.getDomain();
```

Comment:
Why do we need origAppURI here. 

Update deploymentInfo to cache the value of origURI and origAppName in case these gets modified as part of BG deployment.
Further read these values from deploymentInfo and use it in rest of the places.